app/src/main/java/br/com/catbag/gifreduxsample/ui/GifListActivity.java

```diff
@@ -35,36 +33,34 @@ protected void onCreate(Bundle savedInstanceState) {
         mActionCreator.loadGifs();
     }
 
-    private void bindingViews() {
-        //Bindings Defaults
-        Anvil.mount(findViewById(R.id.activity_gif_list), () -> {
-            withId(R.id.loading, () -> {
-                visibility(mGifProgressVisibility);
-            });
-        });
-    }
-
     @Override
     protected Fluxxan<AppState> getFlux() {
         return MyApp.getFluxxan();
     }
 
     @Override
     public void onStateChanged(AppState appState) {
-        mRendered = false;
-        FeedComponent feed = (FeedComponent) findViewById(R.id.feed);
-        feed.setGifs(AppStateHelper.extractGifList(appState));
         if (!appState.getGifs().isEmpty()) {
             mGifProgressVisibility = false;
         }
-
+        //TODO this can be controlled by a singleton class that manage broadcasters and listeners
         ThreadUtils.runOnMain(() -> {
```

Comment:
`        //TODO this can be controlled by a singleton class that manage broadcasters and listeners
        ThreadUtils.runOnMain(() -> {
            Anvil.render();
            if (mAnvilRenderListener != null) mAnvilRenderListener.onAnvilRendered();
        });
`

Esse trecho se repete em todo componente renderizável, talvez valha jogar ele numa super classe.